plat/socionext/synquacer/include/platform_def.h

```diff
@@ -10,13 +10,47 @@
 #include <lib/utils_def.h>
 #include <plat/common/common_def.h>
 
+/* These ARM_* macros are used in Arm scmi driver */
+/*
+ * Macros mapping the MPIDR Affinity levels to ARM Platform Power levels. The
+ * power levels have a 1:1 mapping with the MPIDR affinity levels.
+ */
+#define ARM_PWR_LVL0		MPIDR_AFFLVL0
+#define ARM_PWR_LVL1		MPIDR_AFFLVL1
+#define ARM_PWR_LVL2		MPIDR_AFFLVL2
+#define ARM_PWR_LVL3		MPIDR_AFFLVL3
+
+/*
+ *  Macros for local power states in ARM platforms encoded by State-ID field
+ *  within the power-state parameter.
+ */
+/* Local power state for power domains in Run state. */
+#define ARM_LOCAL_STATE_RUN	U(0)
+/* Local power state for retention. Valid only for CPU power domains */
+#define ARM_LOCAL_STATE_RET	U(1)
+/* Local power state for OFF/power-down. Valid for CPU and cluster power domains */
+#define ARM_LOCAL_STATE_OFF	U(2)
```

Comment:
This is introducing ARM platform layer dependency which might be a maintenance overhead in future. Currently the Power management for SCMI goes through a `css_pm_scmi`  which introduces this dependency. How about removing that dependancy by implementing the functionality in css_pm_scmi.c in synquancer which can then directly invoke SCMI driver ?

I think that would be clean solution.

Reply:
It seems like duplicating stuff. Shouldn't we try to make files under `drivers/arm/css/scp/` generic for reuse in other platforms? Like renaming these macros to PLAT_* rather than ARM_* in common `css_pm_scmi.c`. Or else if they need to be platform specific then move them to `plat/arm`.

Reply:
The `css_scp` layer will have platform implementation details like the power levels available on the platform , the mapping between PSCI power state to SCP power state which makes it difficult to make it generic. The SCMI driver is generic where as `css-scp` is ARM platform specific. 

>  Or else if they need to be platform specific then move them to plat/arm.

Hmm, I see your point. It used to be within `plat/arm` previously and was not intended to be generic. But I agree atleast the `scp` folder should should be within platform directory. The `scmi` folder can go into `drivers/` folder. I will create a task to do that internally.

Reply:
Sounds good.